### PR TITLE
test: test excluding streams threads for uncaught exception handler

### PR DESCRIPTION
**Pull request by wcarlson5**
(cherry picked from commit 6123c42c02d10c41d66f8aa20048763188d08ac3)

Streams no longer sets the java Thread uncaught exception handler and instead uses their own. The issue is once a thread has been replaced it throws the exception to finish killing the thread. Ksql however sets the java Thread uncaught exception handler globally, this calls System.exit(1), perviously the streams handler would override this so the stream threads would be excluded but now that is not the case.

We ignore the calls for the stream threads

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Comment by wcarlson5** [comment]
Integration tests to follow

**Review by stevenpyzhang — APPROVED** [review]
lgtm if jenkins passes

File: ksqldb-rest-app/pom.xml
Comment (agavra) [inline ksqldb-rest-app/pom.xml:183]: nit: all dependencies should be declared in the parent pom and versions should be declared as properties

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/util/KsqlUncaughtExceptionHandler.java
Comment (agavra) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/util/KsqlUncaughtExceptionHandler.java:37]: we might want to create a testing-only constructor here where we can pipe down an action for this if branch. That way we can set a countdown latch or something and make sure that this branch is getting hit instead of relying on sleeps, which both will increase our test run time by a minute and also give us worse coverage

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/util/KsqlUncaughtExceptionHandler.java
Comment (agavra) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/util/KsqlUncaughtExceptionHandler.java:42]: Thanks for addressing the comments! Is there any reason not to use a `CountDownLatch` here instead of a message that we want to print out? I just think that'll be a little cleaner (avoids the assert eventually call and reading to stdout)

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/util/KsqlUncaughtExceptionHandler.java
Comment (stevenpyzhang) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/util/KsqlUncaughtExceptionHandler.java:51]: @agavra I updated the test so that we keep trying to assert that a specific error message that's passed in through the uncaught handler constructor is printed to stderr, which is only done in this logic branch.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/util/KsqlUncaughtExceptionHandler.java
Comment (wcarlson5) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/util/KsqlUncaughtExceptionHandler.java:42]: @stevenpyzhang Wrote this test, not trying to take credit for his work. I just did the 3 line fix. But my local build it messed up. It might be easier for Steven to just push his changes and open a new pr at this point...

**Review by agavra — APPROVED** [review]
Thanks @stevenpyzhang and @wcarlson5 :)

**Review by guozhangwang — APPROVED** [review]
LGTM.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/util/KsqlUncaughtExceptionHandler.java
Comment (stevenpyzhang) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/util/KsqlUncaughtExceptionHandler.java]: it's latched now

**Comment by stevenpyzhang** [comment]
waiting for build to pass will then merge

**Comment by stevenpyzhang** [comment]
FYI, the test takes a while to wrap up in the after part where it stops the server, it spams the logs with ```[2021-04-22 19:06:10,223] WARN [Consumer clientId=_confluent-ksql-default_query_CSAS_S_1-2aca7365-9217-43ba-a8d7-cfdf406ad081-StreamThread-5-consumer, groupId=_confluent-ksql-default_query_CSAS_S_1] Error while fetching metadata with correlation id 2533 : {PAGEVIEW_TOPIC=UNKNOWN_TOPIC_OR_PARTITION} (org.apache.kafka.clients.NetworkClient:1100)``` during this time

**Pull request merged**
